flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/factories/TableFactoryUtil.java

```diff
@@ -164,4 +170,30 @@ public static boolean isLegacyConnectorOptions(
             }
         }
     }
+
+    /** Find and create modification listener list from configuration. */
+    public static List<CatalogModificationListener> findCatalogModificationListenerList(
+            final ReadableConfig configuration, final ClassLoader classLoader) {
```

Comment:
I found 4 usages of this method, but the test case only covers one of them (the `TableEnvironmentImpl` one). What about adding cases for other usages? I'm a bit concern if the `configuration` is passed in correctly end-to-end (like user sets them in SQL client session, TableEnv etc.), instead of being intercepted by a hidden `new Configuration()` that messes everything up.

Reply:
Thanks @PatrickRen for pointing this and I have added tests for all usages for `findCatalogModificationListenerList`